merlin/dag/executors.py

```diff
@@ -332,8 +334,64 @@ def transform(
             )
         )
 
-    def fit(self, dataset: Dataset, graph, strict=False):
-        """Calculates statistics for a set of nodes on the input dataset
+    def fit(self, dataset: Dataset, graph: Graph, schema: Schema = None) -> "Graph":
```

Comment:
Fit was in a weird state where half the code was in NVT and the other half was here in Core. Fitting currently really only exists in NVT, so that wasn't problematic until we needed to test fitting on functionality that only exists in Core. So, this code moved (largely as is) from NVT to Core